src/LanguageServer.Common/Utilities/MSBuildHelper.cs

```diff
@@ -100,15 +100,7 @@ public static void DiscoverMSBuildEngine(string baseDirectory = null, ILogger lo
                 build: targetSdkSemanticVersion.Patch
             );
 
-            var queryOptions = new VisualStudioInstanceQueryOptions
-            {
-                // We can only load the .NET Core MSBuild engine
-                DiscoveryTypes = DiscoveryType.DotNetSdk
-            };
-
-            VisualStudioInstance[] allInstances = MSBuildLocator
-                .QueryVisualStudioInstances(queryOptions)
-                .ToArray();
```

Comment:
`QueryVisualStudioInstances` already searches with `DiscoveryType.DotNetSdk` by default. `ToArray` is also unnecessary since wi apply other linq methods on `allInstances` anyway

Reply:
The `.ToArray()` call is useful for debugging scenarios (to see which instances are returned before further filtering is applied).

Reply:
You can still see IEnumerable's values, at least in VS:
![devenv_CJslhQBlEx](https://github.com/tintoy/msbuild-project-tools-server/assets/70431552/14f23df9-098b-4621-b3bb-24cce92d5903)
Morover, VS for some time now has a dedicated IEnumerable visualizer window:
![devenv_dHd1l4qxif](https://github.com/tintoy/msbuild-project-tools-server/assets/70431552/b5336f93-7393-4143-9e4d-8e84c5bada20)
So this is not a concern for me)